core/src/main/java/org/elasticsearch/search/suggest/Suggest.java

```diff
@@ -399,35 +400,17 @@ public XContentBuilder toXContent(XContentBuilder builder, Params params) throws
                         "Cannot parse suggestion response without type information. Set [" + RestSearchAction.TYPED_KEYS_PARAM
                                 + "] parameter on the request to ensure the type information is added to the response output");
             }
-            Suggestion suggestion = null;
-            Function<XContentParser, Entry> entryParser = null;
-            // the "size" parameter and the SortBy for TermSuggestion cannot be parsed from the response, use default values
-            // TODO investigate if we can use NamedXContentRegistry instead of this switch
-            switch (type) {
-                case Suggestion.NAME:
```

Comment:
Should we keep support for this "generic" suggestion type as long as it can (in theory) still be instantiated? I'd be in favour of making it abstract and removing it from the transport layer (including support for its use in Custom Suggester) on master in a follow up PR after re-checking with @areek that we realy don't need it.